### ISSUE TITLE
Got error on Helper.php#L190

### ISSUE THREAD

**Issue by Gounlaf** [yajra/laravel-datatables#108]
Hi,

Recent changes on Helper.php introduce bug: I got an ErrorException "Trying to get property of non-object"

How it happens :

```php
$collection = new Collection();
$collection->push(array(
    'object_value' => \Carbon\Carbon::now()
));

Datatables::of($collection)->make(true);
```

```
in Helper.php line 192
at HandleExceptions->handleError('8', 'Trying to get property of non-object', '/Users/florianlevis/Server/Ridemore/ridemore-rental-platform/vendor/yajra/laravel-datatables-oracle/src/yajra/Datatables/Helper.php', '192', array('row' => array('object_value' => object(Carbon)), 'data' => array('object_value' => object(Carbon)), 'key' => 'object_value')) in Helper.php line 192
at Helper::convertToArray(array('object_value' => object(Carbon))) in DataProcessor.php line 72
at DataProcessor->process(true) in BaseEngine.php line 622
at BaseEngine->render(true) in BaseEngine.php line 528
at BaseEngine->make(true, true) in CollectionEngine.php line 191
at CollectionEngine->make(true) in SkateController.php line 1024
```

Code:

```php
    public static function convertToArray($row)
    {
        $data = $row instanceof Arrayable ? $row->toArray() : (array) $row;
        foreach (array_keys($data) as $key) {
            if (is_object($data[$key])) {
                // Here, $data[$key] is object, but $row not necessary (that's why you do the cast)
                // => It should be $data[$key] = $data[$key], because you are casting $row
                // and so there is nothing to do
                // I don't understand exactly what you want to do =/
                $data[$key] = $row->$key;
            } else {
                if (is_array($data[$key])) {
                    $data[$key] = static::convertToArray($data[$key]);
                }
            }
        }

        return $data;
    }
```

Regards,

**Comment by yajra** [comment]
@Gounlaf, thanks for reporting, updates here is related to fix eager loading on v5.5.6. However, after fixing it an error indeed occur on objects value like carbon. That is what I'm trying to fix on v5.5.7 but as reported I think it's not fully fixed.

**Comment by Gounlaf** [comment]
OK. Let me know if I can help you ...

**Comment by yajra** [comment]
Yes, please. If you have time, it would be great if you can send a PR to fix this. But I'm also looking on this issue now. Thanks for offering a help. ^_^

**Comment by Gounlaf** [comment]
* What the goals of this method ? Convert anything in array ?
* "eager loading" => so, need to "call/read" the property in order to load the data ?

**Comment by yajra** [comment]
Yes, convert all to associative array. If the value is an object, convert all properties to array. If value is like an eager loaded relationship, we also need to convert it to array.

**Comment by Gounlaf** [comment]
OK ; I will try to do a PR

**Comment by yajra** [comment]
This function is mainly use on `DataProcessor` - `process` function. Basically, preparing the final output to display.

**Comment by yajra** [comment]
@Gounlaf, i think we just need to call the function again for object values. Can you please test this function on your code? I don't have an existing use case like yours.

```php
public static function convertToArray($row)
    {
        $data = $row instanceof Arrayable ? $row->toArray() : (array) $row;
        foreach (array_keys($data) as $key) {
            if (is_object($data[$key])) {
                $data[$key] = self::convertToArray($data[$key]);
            } else {
                if (is_array($data[$key])) {
                    $data[$key] = self::convertToArray($data[$key]);
                }
            }
        }

        return $data;
    }
```

**Comment by Gounlaf** [comment]
It's working ; so you can simplify the code now

**Comment by yajra** [comment]
Oh yeah, we can simplify the code. Can you please test again the simplified code? Then will push it later after testing on my demo app.

```php
public static function convertToArray($row)
    {
        $data = $row instanceof Arrayable ? $row->toArray() : (array) $row;
        foreach (array_keys($data) as $key) {
            if (is_object($data[$key]) || is_array($data[$key])) {
                $data[$key] = self::convertToArray($data[$key]);
            }
        }

        return $data;
    }
```

**Comment by yajra** [comment]
@Gounlaf, thanks a lot for your help. Should be fixed now on v5.5.8

**Comment by Gounlaf** [comment]
@yajra sorry, was AFK ; It's working for me now ; thanks !
